aten/src/ATen/native/Distributions.cpp

```diff
@@ -209,13 +209,17 @@ Tensor _s_gamma_cpu(const Tensor& alpha, Generator *gen) {
     std::lock_guard<std::mutex> lock(generator->mutex);
     CPU_tensor_apply2<scalar_t, scalar_t>(ret, alpha,
       [generator](scalar_t& ret_val, const scalar_t& alpha){
-        BaseSampler<double> standard_uniform([generator] () {
+
+        auto uniform_lambda = [generator] () {
```

Comment:
nit: Would be even better with a comment saying that you can't inline the lambda to workaround hcc bug, with a link to the upstream hcc bug report :>

Reply:
@ezyang It's actually not a bug, but rather ROCM doesn't have nvfunctional equivalent